components/uniset/src/uniset.rs

```diff
@@ -0,0 +1,307 @@
+use std::{
+    char::{from_u32, MAX},
+    convert::TryFrom,
+    ops::RangeBounds,
+    slice::Iter,
+};
+
+use crate::utils::{deconstruct_range, is_sorted};
+// use crate::UnicodeSetIter;
+/// Represents the end code point of the Basic Multilingual Plane range, starting from code point 0 , inclusive
+const BMP_MAX: u32 = 0xFFFF;
+
+/// UnicodeSet membership wrapper
+///
+/// Provides exposure to membership functions and constructors from serialized UnicodeSets
+/// and predefined ranges.
+/// Implements an inversion list.
+//#[derive(Copy, Clone, Debug, Eq)]
+pub struct UnicodeSet {
+    // If we wanted to use an array to keep the memory on the stack, there is an unsafe nightly feature
+    // https://doc.rust-lang.org/nightly/core/array/trait.FixedSizeArray.html
+    // Allows for traits of fixed size arrays
+    inv_list: Vec<u32>,
+}
+
+impl TryFrom<Vec<u32>> for UnicodeSet {
+    type Error = String;
+
+    fn try_from(set: Vec<u32>) -> Result<Self, Self::Error> {
+        if is_sorted(&set) {
```

Comment:
You should also check that the inversion list is in range.  For example, no values greater than char::MAX. 
 Maybe also check that the array has an even length?

Reply:
agreed and changed, this was originally implemented in the `parse_serial_string` method but I forgot to include it when transitioning to `try_from::<Vec<u32>>`